### ISSUE TITLE
Content-Location header in HTTP::Response [rt.cpan.org #76578]

### ISSUE THREAD

**Issue by oalders** [libwww-perl/HTTP-Message#51]
Migrated from [rt.cpan.org#76578](https://rt.cpan.org/Ticket/Display.html?id=76578) (status was 'new')

Requestors:
* perl@toby.ink

From perl@toby.ink on 2012-04-15 21:52:51:
```
Although RFC 2616 does say that Content-Location sets the base URI for a 
response, browsers have never implemented it that way. As a result, in 
the forthcoming update to HTTP 1.1 this is being changed, and Content-
Location should no longer set the base URI.

See http://trac.tools.ietf.org/wg/httpbis/trac/ticket/154

I'd like to request HTTP::Response's "base" method no longer takes 
Content-Location into account, or some sort of option be provided to 
allow it to be ignored.
```

**Comment by neilb** [comment]
[RFC 7231](https://www.rfc-editor.org/rfc/rfc7231) says

> The definition of Content-Location has been changed to no longer
   affect the base URI for resolving relative URI references, due to
   poor implementation support and the undesirable effect of potentially
   breaking relative links in content-negotiated resources.

If you're happy to go with this, I think it just needs:

 * removal of [this line of code](https://metacpan.org/dist/HTTP-Message/source/lib/HTTP/Response.pm#L87)
 * rewording of [this line of doc](https://metacpan.org/dist/HTTP-Message/source/lib/HTTP/Response.pm#L485)
 * change [this test](https://metacpan.org/release/OALDERS/HTTP-Message-6.41/source/t/response.t#L107-108) to confirm that setting the Content-Location doesn't affect what `base()` returns.

There's a chance this may break things downstream that are relying on this behaviour. How seriously do you want to take that possibility?

**Comment by oalders** [comment]
I think it's ok to risk something breaking, given that this change was implemented 13 years ago. If it breaks something downstream, we could look at implementing some kind of fallback. I'll see if anyone else has an opinion.

**Comment by oalders** [comment]
Closed via #185.
